lxd/devices.go

```diff
@@ -549,65 +595,56 @@ func deviceTaskBalance(s *state.State) {
 				logger.Warn("The pinned CPUs override the NUMA configuration CPUs", logger.Ctx{"pinnedCPUs": instanceCpus, "numaCPUs": numaCpus})
 			}
 
-			fillFixedInstances(fixedInstances, c, cpus, instanceCpus, len(instanceCpus), false)
+			usableCpus := cpus
+			//
+			// For VM instances, historically, we allow explicit pinning to a CPUs
```

Comment:
what happens if you try to explicitly pin a container's cpu to one in the isolcpus list, does the container refuse to start?

Reply:
Nope, historically, it will just silently ignore such pinning and produce a warning in the logs `Internal error: instance using unavailable cpu`. I'm not changing behavior for containers here for compatibility reasons.

Reply:
OK thanks. Can we update the logged message to use contextual logging (i.e not `logger.Errorf`) and include the instance name, project name and, perhaps CPU ID?

Reply:
I wasn't fully right when I said that this incorrect container pinning produces `Internal error: instance using unavailable cpu`. Actually, this error is actually internal and it's not easy to get it. But what I did is added a new:
```
				logger.Warn("Instance using unavailable cpu", logger.Ctx{"project": inst.Project().Name, "instance": inst.Name(), "cpu": nr})
```
in the `fillFixedInstances` function.

Reply:
thanks!